code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -26,10 +26,12 @@
 	var/datum/action/innate/slime_pick_up/slime_up_action = new
 	var/datum/action/innate/feed_slime/feed_slime_action = new
 	var/datum/action/innate/monkey_recycle/monkey_recycle_action = new
+	var/datum/action/innate/transfer_potion/transfer_potion_action = new
 
 	var/list/stored_slimes = list()
 	var/max_slimes = 5
 	var/monkeys = 0
+	var/obj/item/slimepotion/current_potion
```

Comment:
This isn't going to work well.

You need to subtype the potions that you actually can use on slimes ensure only those cna be slotted into the machine.

Not sense in being able to put a speed potion into the machine and attempt to apply it to a slime when it doesn't do anything.

Reply:
I wanted to be a little more OOP and allow for future slime affecting potions to be added without snowflakes. Perhaps add a variable to the slime potion type that allows it to be inserted?

Reply:
What about a list of potions that are accepted instead? Wouldn't that be a better way than a var?

Reply:
I guess yea

Reply:
just doing what TG did by making slime potions that can be used `/obj/item/slime_potion/slime` avoids the snowflake and makes it completely OOP.

Reply:
Just checked TG's code, `else if(istype(O, /obj/item/slimepotion/slime))` is in the attack proc for the slime console and every single slime option, even one's that do nothing on slimes, are of this type.